.github/workflows/ci.yaml

```diff
@@ -0,0 +1,108 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  devcontainer:
+    # Deduplicate jobs from pull requests and branch pushes within the same repo.
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
```

Comment:
You could have this logic in a `check` job and then use `needs: check`, rather than duplicating it.

Reply:
That'd be a neat way of doing so, do you think it'd be more readable?

Reply:
I think so, it just removes a bit of cognitive complexity when looking at a specific job. Plus then then check can be updated easily if some other case is required.

Reply:
Actually, is this the place for YAML anchors? E.g.

```yaml
definitions:
    # Deduplicate jobs from pull requests and branch pushes within the same repo.
    deduplicated: &deduplicated
        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository

jobs:
    devcontainer:
        <<: *deduplicated
        uses: ./.github/workflows/_devcontainer.yaml

    bundler_container:
        <<: *deduplicated
        uses: ./.github/workflows/_bundler_container.yaml
```

Reply:
Never mind, GitHub actions don't support YAML anchors

Reply:
After trying a `deduplicate` job with `needs: deduplicate` on others, we have opted to stay with too many `if`s. Mainly due to the added mess caused in the checks DAG